### PR TITLE
Add simplifications for List.sort, List.sortBy and List.sortWith

### PR DESCRIPTION
**Pull request by lue-bird**
```elm
List.sortBy identity list
--> list

List.sortBy (\_ -> a) list
--> list

List.sortWith (\_ _ -> LT) list
--> List.reverse list

List.sortWith (\_ _ -> EQ) list
--> list

List.sortWith (\_ _ -> GT) list
--> list

-- The following simplifications for List.sort also work for List.sortBy fn and List.sortWith fn
List.sort []
--> []

List.sort [ a ]
--> [ a ]
```
The first 22 commits are from merging the branch from the previous PR with main. If you want a clean record in this PR, I'll close and reopen with a fresh branch (something something cherry-pick).

You said that you prefer smaller PRs. Are we still to big? Would checks for one function = one PR be better?

### CODE REVIEW

**Review by jfmengels — COMMENTED** [review]
Will review the rest later!

File: src/Simplify.elm [hunk 1]
```diff
@@ -4133,6 +4159,220 @@ listReverseChecks ({ parentRange, fnRange, firstArg } as checkInfo) =
...
+                                checkInfo.fnRange
+                                [ Fix.removeRange
+                                    { start = checkInfo.parentRange.start
+                                    , end = listArgument.start
+                                    }
+                                ]
+                            ]
+            in
+            if isIdentity checkInfo.lookupTable checkInfo.firstArg then
+                fixToIdentity "identity"
```
[the remark this answers is not in the document]
Comment (lue-bird) [reply src/Simplify.elm]: Yes that's already fixed.
Should we still add `sortBy identity --> sort`?

**Comment by jfmengels** [comment]
Very good work @lue-bird :relaxed: 

To answer your questions (which I forgot about): I do prefer nicely rebased PRs yeah. That's always nicer. I don't know how comfortable you feel with rebasing though.
In terms of contents, I would prefer smaller PRs yeah. For instance, only `List.sort` or only `List.sortBy`. Or if a lot of changes are there, maybe a subfraction. That way, it's easier to approve smaller bits and get them in. The downside is that it requires more rebasing work from you unfortunately :/

File: src/Simplify.elm (hunk 1 above)
[the remark this answers is not in the document]
Comment (jfmengels) [reply src/Simplify.elm]: Yes we should :+1:

**Pull request merged**